CMake/external_foonathan_memory.cmake

```diff
@@ -1,83 +1,44 @@
 cmake_minimum_required(VERSION 3.11) 
 include(FetchContent)
 
-# Mark new options from FetchContent as advanced options
-mark_as_advanced(FETCHCONTENT_QUIET)
-mark_as_advanced(FETCHCONTENT_BASE_DIR)
-mark_as_advanced(FETCHCONTENT_FULLY_DISCONNECTED)
-mark_as_advanced(FETCHCONTENT_UPDATES_DISCONNECTED)
+# We use a function to enforce a scoped variables creation only for FastDDS build (i.e turn off BUILD_SHARED_LIBS which is used on LRS build as well)
+function(get_foonathan_memory)
 
-message(CHECK_START  "Fetching & Installing foonathan_memory...")
-list(APPEND CMAKE_MESSAGE_INDENT "  ")
+    # Mark new options from FetchContent as advanced options
+    mark_as_advanced(FETCHCONTENT_QUIET)
+    mark_as_advanced(FETCHCONTENT_BASE_DIR)
+    mark_as_advanced(FETCHCONTENT_FULLY_DISCONNECTED)
+    mark_as_advanced(FETCHCONTENT_UPDATES_DISCONNECTED)
 
-FetchContent_Declare(
-  foonathan_memory
-  GIT_REPOSITORY https://github.com/foonathan/memory.git
-  GIT_TAG        "v0.7-2"
-  GIT_SHALLOW ON    # No history needed
-  SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/third-party/foonathan_memory
-)
+    message(CHECK_START  "Fetching foonathan_memory...")
+    list(APPEND CMAKE_MESSAGE_INDENT "  ")
 
-# Remove unrequired targets
-set(FOONATHAN_MEMORY_BUILD_VARS -DBUILD_SHARED_LIBS=OFF             # explicit set static lib
-                                -DFOONATHAN_MEMORY_BUILD_EXAMPLES=OFF 
-                                -DFOONATHAN_MEMORY_BUILD_TESTS=OFF
-                                -DFOONATHAN_MEMORY_BUILD_TOOLS=ON)  # this tool is needed during configure time only, FastDDS recommend turning it ON.
-   
-# Align STATIC CRT definitions with LRS   
-if(BUILD_WITH_STATIC_CRT)
-    set(FOONATHAN_MEMORY_BUILD_VARS ${FOONATHAN_MEMORY_BUILD_VARS}
-                                    -DCMAKE_POLICY_DEFAULT_CMP0091:STRING=NEW 
-                                    -DCMAKE_MSVC_RUNTIME_LIBRARY:STRING=MultiThreaded$<$<CONFIG:Debug>:Debug>)
-endif()  
-  
+    FetchContent_Declare(
+      foonathan_memory
+      GIT_REPOSITORY https://github.com/foonathan/memory.git
+      GIT_TAG        "v0.7-3"
+      GIT_SHALLOW ON    # No history needed
+      SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/third-party/foonathan_memory
```

Comment:
Did you try static CRT vs no static CRT?
```
if(BUILD_WITH_STATIC_CRT)
    set(FOONATHAN_MEMORY_BUILD_VARS ${FOONATHAN_MEMORY_BUILD_VARS}
```

Reply:
I did, it made no difference.
I'll try again.

Reply:
All combinations work and link fine. When I look at the settings for foonathan, when the flag is off it's set to "Multi-threaded DLL (/MD)" and when on it's "Multi-threaded (/MT)".